src/main/java/com/notably/logic/commands/DeleteCommand.java

```diff
@@ -20,13 +23,17 @@ public DeleteCommand(Path targetPath) {
     }
 
     @Override
-    public void execute(Model notablyModel) {
-        requireNonNull(notablyModel);
-        if (this.targetPath instanceof RelativePath) {
-            RelativePath toConvert = (RelativePath) this.targetPath;
-            this.targetPath = toConvert.toAbsolutePath(notablyModel.getCurrentlyOpenPath());
+    public void execute(Model notablyModel) throws CommandException {
+        try {
+            requireNonNull(notablyModel);
+            if (this.targetPath instanceof RelativePath) {
+                RelativePath toConvert = (RelativePath) this.targetPath;
+                this.targetPath = toConvert.toAbsolutePath(notablyModel.getCurrentlyOpenPath());
+            }
```

Comment:
Let's move this section out of the try-catch block for clarity! I believe it makes clearer that it's the code below (`removeBlock`) that might throw the errors you'd like to catch

Reply:
done!